### ISSUE TITLE
Summer time issues

### ISSUE THREAD

**Issue by CGenie** [FossifyOrg/Clock#61]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Clock/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Clock/issues) or [closed](https://github.com/FossifyOrg/Clock/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Clock/discussions) or [closed](https://github.com/FossifyOrg/Clock/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Clock/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.0

### Affected Android/Custom ROM version

EMUI 12.0.0

### Affected device model

Huawei P40 Lite

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

This is described here: https://github.com/FossifyOrg/Clock/discussions/60.

1. I set alarm to 6 AM.
2. Summer time change happened
3. The alarm fires at 7 AM even though it's reported as 6 AM in the app.

### Expected behavior

Ring at 6 AM as stated.

### Actual behavior

Rings at 7 AM.

**Comment by Beanow** [comment]
See also https://github.com/SimpleMobileTools/Simple-Clock/issues/447

**Comment by GuitarBilly** [comment]
I can confirm this bug behaviour. I had to toggle all my alarms off and on to get them re-synced after dst change.

Clock 1.0.0
Android 7.1.2
Nokia 6 - TA-1033

**Comment by harmenkets** [comment]
I can confirm there is still an issue with daylight saving time (DST).
My country (NL) just switched to wintertime last night. The time on my device shifted correctly, but my sunday morning alarm went off 1 hour too early *and* on the (new) correct time (so it went off twice while I had set only 1 alarm).

Clock: 1.0.0
Android: 14 AP2A.240805.005
Make/Model: Xiaomi Mi Pad 4 (clover)

**Comment by tomcodemusic** [comment]
Can confirm as well. My weekday alarm went off one hour too early, while the device itself had the correct time since the DST switch yesterday. When I then disabled and enabled the same alarm again (without any other changes), it went off at the correct time.

Clock: 1.0.0
Android: 14
Device: Samsung Galaxy A52s

**Comment by CGenie** [comment]
I didn't touch anything in the app and it keeps ringing twice, i.e. for an alarm set at 15, it rings at 14 and then at 15.  I had 2 alarms set for yesterday and 1 for today, and overall there were 6 ringings :)

**Comment by alvaroemtnez** [comment]
Hello,

I experienced the same bug today. I have an alarm set from monday to friday at 7:00, and today it rang at 6:00. The winter time change was yesterday at 3:00 so I didn't notice until today (didn't have an alarm on Sunday).

Disabling and enabling the alarms seems to solve it (no need to delete them), but it's difficult to remember to do that as this happens only twice a year. On the summer change that could mean 1hr late to work :(

Clock: 1.0.0
Android: 15 (GrapheneOS 2024102400)
Device: Google Pixel 8

**Comment by brandje41** [comment]
Bug confirmed as well.  

Clock: 1.0.0
Android: 14 (Samsung default)
Device: Samsung Galaxy 23

**Comment by Gouvernathor** [comment]
Sameeeee
and in my experience it rang twice : 7AM (bug) and then 8AM.

**Comment by Joshix-1** [comment]
made a pr, would be cool if someone could test and review it. Would be good if it was fixed before the next timezone change 🤣

**Comment by naveensingh** [comment]
It should be fixed in the next update but for those who would like to try it now, a test APK is [linked here](https://github.com/FossifyOrg/Clock/issues/73#issuecomment-2708934417).

**Comment by naveensingh** [comment]
Received a report from someone on 1.1.0 that this may not be fixed or at least not fixed properly.

**Comment by Joshix-1** [comment]
My alarm went off correctly today and yesterday. (Clocks switched to summertime yesterday)

**Comment by Joshix-1** [comment]
Maybe the update was installed after the alarm was already scheduled. That could be a problem. If the alarm is only going off on weekdays and the app was updated on the weekend that could probably cause this problem as the alarm was scheduled using the old logic

**Comment by naveensingh** [comment]
> Maybe the update was installed after the alarm was already scheduled. That could be a problem. If the alarm is only going off on weekdays and the app was updated on the weekend that could probably cause this problem as the alarm was scheduled using the old logic

Good catch. I can confirm that the alarms scheduled using 1.0.0 still fire at incorrect times when app is updated to 1.1.0. Alarms scheduled using 1.1.0 or master work as expected. 

The ship has sailed but adding another receiver to reschedule alarms on app update, it should help in the future.
